### PR TITLE
Fix issues #10300, #10286, #10280, #10214

### PR DESCRIPTION
**Pull request by kavishkafernando**
Fixes https://github.com/wso2/product-apim/issues/10300
Fixes https://github.com/wso2/product-apim/issues/10286
Fixes https://github.com/wso2/product-apim/issues/10280
Fixes https://github.com/wso2/product-apim/issues/10214

### CODE REVIEW

**Review by malinthaprasan — APPROVED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx [hunk 1]
```diff
@@ -126,41 +126,74 @@ const useStyles = makeStyles((theme) => ({
...
+            securityScheme: [...api.securityScheme],
+            keyManagers: [...(api.keyManagers || [])],
+            corsConfiguration: {
+                corsConfigurationEnabled: api.corsConfiguration.corsConfigurationEnabled,
+                accessControlAllowCredentials: api.corsConfiguration.accessControlAllowCredentials,
+                accessControlAllowOrigins: [...api.corsConfiguration.accessControlAllowOrigins],
+                accessControlAllowHeaders: [...api.corsConfiguration.accessControlAllowHeaders],
+                accessControlAllowMethods: [...api.corsConfiguration.accessControlAllowMethods],
+            },
+            apiwebsubSubscriptionConfiguration: {
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx:187]: Can't we separate out the API product-related config params from the API like `apiwebsubSubscriptionConfiguration` and prevent repeating the same properties twice in the API and API product config JSON?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx [hunk 2]
```diff
@@ -298,11 +331,13 @@ export default function RuntimeConfiguration() {
     const { api, updateAPI } = useContext(APIContext);
     const isAsyncAPI = api.type === 'WS' || api.type === 'WEBSUB' || api.type === 'SSE';
     const isWebSub = api.type === 'WEBSUB';
-    api.websubSubscriptionConfiguration = api.websubSubscriptionConfiguration || {
-        secret: '',
-        signingAlgorithm: '',
-        signatureHeader: '',
-    };
+    if (!api.isAPIProduct()) {
+        api.websubSubscriptionConfiguration = api.websubSubscriptionConfiguration || {
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx:335]: I think this `api` object is coming from the APIContext , so altering it directly is not valid

**Review by tmkasun — APPROVED** [review]
Please address the comments and merge

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx (hunk 1 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx]: fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx (hunk 2 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Configuration/RuntimeConfiguration.jsx]: removed

**Pull request merged**